### PR TITLE
feat(localization): Add strings to string file [IN-1070]

### PR DESCRIPTION
**Pull request by ckpocket**
feat(localization): Added localizable strings in English string [IN-1070]

## Summary
- Using localized string for strings in code
- For english only, unit & UI Tests are working as expected
- App can be run in another language supported by pocket app.

## References 
https://getpocket.atlassian.net/wiki/spaces/CP/pages/2820603915/PRD+-+Localization

## Implementation Details
- Removed localization from Root folder
- Added localization string files in Pocket Kit package

## Test Steps
- The feature can not be tested by QA since for other languages, sting files aren't translated yet.
- However, to test it, you can checkout the branch, add dummy strings in French (whichever your preferred) and run the app locally in Xcode.

## PR Checklist:
- [X] Added Unit / UI tests
- [X] Self Review (review, clean up, documentation, run tests)
- [X] Basic Self QA

[IN-1070]: https://getpocket.atlassian.net/browse/IN-1070?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Pocket.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -43,10 +43,8 @@
 		16E4B16F27A3212900E01746 /* Response+factories.swift in Sources */ = {isa = PBXBuildFile; fileRef = 16E4B16E27A3212900E01746 /* Response+factories.swift */; };
 		16EE3F5026CEC80900249AF4 /* PullToRefreshTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 16EE3F4F26CEC80900249AF4 /* PullToRefreshTests.swift */; };
 		16FCADBF26CB04AC00906C03 /* ArchiveAnItemTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 16FCADBE26CB04AC00906C03 /* ArchiveAnItemTests.swift */; };
-		3E15B8F0292688EB009A26B6 /* Localizable.strings in Resources */ = {isa = PBXBuildFile; fileRef = 3E15B8F2292688EB009A26B6 /* Localizable.strings */; };
```
Comment (cyndichin) [inline Pocket.xcodeproj/project.pbxproj:46]: out of curiosity, why are these removed?

File: PocketKit/Sources/PocketKit/String+Localization.swift [hunk 2]
```diff
@@ -2,6 +2,10 @@ import Foundation
 
 extension String {
     func localized(withComment: String = "") -> String {
-        return NSLocalizedString(self, tableName: nil, bundle: Bundle.main, value: "Localizable string not found!", comment: withComment)
+        return NSLocalizedString(self, tableName: nil, bundle: .module, value: "Localizable string not found!", comment: withComment)
+    }
+
+    func localized(_ withArguments: CVarArg..., comment: String = "") -> String {
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/String+Localization.swift:8]: in which case do we use this vs the other `localized` function

**Review by cyndichin — COMMENTED** [review]
looks good, a couple of comments for my clarity. Also according to the ticket, I thought we would be consolidating our hard coded strings in one place. Is this no longer true?

Also, how do we decided what strings to localize or not? Will we have to use `localized()` everywhere in the future?

File: Pocket.xcodeproj/project.pbxproj (hunk 1 above)
Comment (ckpocket) [reply Pocket.xcodeproj/project.pbxproj:46]: These are moved to PocketKit instead of at root level

File: PocketKit/Sources/PocketKit/String+Localization.swift (hunk 2 above)
Comment (ckpocket) [reply PocketKit/Sources/PocketKit/String+Localization.swift:8]: If you would like to pass an argument:
EX: %d min read".localized(timeToRead)

File: PocketKit/Sources/PocketKit/Home/RecentSavesItemCell.swift [hunk 3]
```diff
@@ -55,7 +55,7 @@ class RecentSavesItemCell: HomeCarouselItemCell {
                 return nil
             }
 
-            return "\(timeToRead) min read"
+            return "%d min read".localized(timeToRead)
```
Comment (ckpocket) [inline PocketKit/Sources/PocketKit/Home/RecentSavesItemCell.swift:58]: @cyndichin Usage of localized with argument here!

**Comment by ckpocket** [comment]
> 

- Yes, we have to localized() wherever localized string is need. Assuming string is added .string file.
- Strings are hardcoded in .string files. All files are in this PR.

File: PocketKit/Sources/PocketKit/String+Localization.swift (hunk 2 above)
Comment (cyndichin) [reply PocketKit/Sources/PocketKit/String+Localization.swift:8]: thanks for the clarification!

**Comment by cyndichin** [comment]
> * Yes, we have to localized() wherever localized string is need. Assuming string is added .string file.
> * Strings are hardcoded in .string files. All files are in this PR.

It seems that we aren't pulling the strings from one file though, so going forward would we have to copy the hardcoded string from the codebase into these files?

**Review by cyndichin — APPROVED** [review]
code itself looks good, so approving. We can continue the convo here:
https://pocket.slack.com/archives/CBKLKAUTC/p1674576836337359

Seems like Bitrise is failing on the sentry issue, a rebuild should fix it.

**Review by cyndichin — APPROVED** [review]

**Review by cyndichin — CHANGES_REQUESTED** [review]
do we want to add this to a `feature branch` instead of merging to develop?

**Comment by Gio2018** [comment]
> do we want to add this to a `feature branch` instead of merging to develop?

Hey @cyndichin , what would be the pros/cons of adding a feature branch vs merging into develop?

**Review by cyndichin — APPROVED** [review]

**Comment by cyndichin** [comment]
> > do we want to add this to a `feature branch` instead of merging to develop?
> 
> Hey @cyndichin , what would be the pros/cons of adding a feature branch vs merging into develop?

Good question, this is what was discussed: https://pocket.slack.com/archives/CBKLKAUTC/p1674512857046319?thread_ts=1674487587.669919&cid=CBKLKAUTC

But after our conversation in the morning, I don't see any high risk in merging this work.
